docs/tutorials/gluon/learning_rate_finder.md

```diff
@@ -0,0 +1,321 @@
+from mxnet.gluon.data.vision import transforms
+
+transform = transforms.Compose([
+    # Switches HWC to CHW, and converts to `float32`
+    transforms.ToTensor(),
+    # Channel-wise, using pre-computed means and stds 
+    transforms.Normalize(mean=[0.4914, 0.4822, 0.4465],
+                         std=[0.2023, 0.1994, 0.2010])
+])
+
+dataset = mx.gluon.data.vision.datasets.CIFAR10(train=True).transform_first(transform)
+
+class ContinuousBatchSampler():
+    def __init__(self, sampler, batch_size):
+        self._sampler = sampler
+        self._batch_size = batch_size
+
+    def __iter__(self):
+        batch = []
+        while True:
+            for i in self._sampler:
+                batch.append(i)
+                if len(batch) == self._batch_size:
+                    yield batch
+                    batch = []
+
+sampler = mx.gluon.data.RandomSampler(len(dataset))
+batch_sampler = ContinuousBatchSampler(sampler, batch_size=128)
+data_loader = mx.gluon.data.DataLoader(dataset, batch_sampler=batch_sampler, num_workers=cpu_count())
+```
+
+## Implementation
+
+With preparation complete, we're ready to write our Learning Rate Finder that wraps the `Learner` we defined above. We implement a `find` method for the procedure, and `plot` for the visualization. Starting with a very low learning rate as defined by `lr_start` we train one iteration at a time and keep multiplying the learning rate by `lr_multiplier`. We analyse the loss and continue until it diverges according to `LRFinderStoppingCriteria` (which is defined later on). You may also notice that we save the parameters and state of the optimizer before the process and restore afterwards. This is so the Learning Rate Finder process doesn't impact the state of the model, and can be used at any point during training.
+
+
+```python
+from matplotlib import pyplot as plt
+
+class LRFinder():
+    def __init__(self, learner):
+        """
+        learner: able to take single iteration with given learning rate and return loss
+           and save and load parameters of the network (Learner)
+        """
+        self.learner = learner
+        
+    def find(self, lr_start=1e-6, lr_multiplier=1.1, smoothing=0.3):
+        """
+        lr_start: learning rate to start search (float)
+        lr_multiplier: factor the learning rate is multiplied by at each step of search (float)
+        smoothing: amount of smoothing applied to loss for stopping criteria (float)
+        """
+        # Used to initialize weights; pass data, but don't take step.
+        # Would expect for new model with lazy weight initialization
+        self.learner.iteration(take_step=False)
+        # Used to initialize trainer (if no step has been taken)
+        if not self.learner.trainer._kv_initialized:
+            self.learner.trainer._init_kvstore()
+        # Store params and optimizer state for restore after lr_finder procedure
+        self.learner.net.save_params("lr_finder.params")
```

Comment:
"save_params" is deprecated. Should we use save_parameters instead?

Reply:
Changed.